gensim/similarity_learning/sl_vocab.py

```diff
@@ -0,0 +1,557 @@
+from collections import Counter
+import pandas as pd
+import numpy as np
+import logging
+import re
+import random
+
+logger = logging.getLogger(__name__)
+
+
+"""
+This file contains the WikiQAExtractor, WikiQA_DRMM_TKS_Extractor and QuoraQPExtractor which are utility classes to
+extract data from the data files.
+"""
+
+
+class WikiQA_DRMM_TKS_Extractor:
```

Comment:
To be honest, this class should be partially "hidden" (see how `Word2Vec` implemented)